rust/tests/datafusion_test.rs

```diff
@@ -249,30 +272,57 @@ async fn test_files_scanned() -> Result<()> {
     let (_temp_dir, table) = prepare_table(batches, SaveMode::Append).await;
     assert_eq!(table.version(), 2);
 
-    let ctx = SessionContext::new();
-    let plan = table.scan(&ctx.state(), None, &[], None).await?;
-    let plan = CoalescePartitionsExec::new(plan.clone());
+    let metrics = get_scan_metrics(&table, &state, &[]).await?;
+    assert!(metrics.num_scanned_files() == 2);
 
-    let task_ctx = Arc::new(TaskContext::from(&ctx.state()));
-    let _ = collect(plan.execute(0, task_ctx)?).await?;
+    let e = col("id").gt(lit(5));
+    let metrics = get_scan_metrics(&table, &state, &[e]).await?;
+    assert!(metrics.num_scanned_files() == 1);
 
-    let mut metrics = ExecutionMetricsCollector::default();
-    visit_execution_plan(&plan, &mut metrics).unwrap();
+    // Ensure that tables without stats and partition columns can be pruned for just partitions
+    let table = deltalake::open_table("./tests/data/delta-0.8.0-null-partition").await?;
+
+    /*
+    // Logically this should prune... Might require an update on datafusion
+    let e = col("k").eq(lit("A")).and(col("k").is_not_null());
+    let metrics = get_scan_metrics(&table, &state, &[e]).await.unwrap();
+    println!("{:?}", metrics);
+    assert!(metrics.num_scanned_files() == 1);
+
+    let e = col("k").eq(lit("B"));
+    let metrics = get_scan_metrics(&table, &state, &[e]).await?;
+    assert!(metrics.num_scanned_files() == 1);
+    */
+
+    // Check pruning for null partitions
+    let e = col("k").is_null();
+    let metrics = get_scan_metrics(&table, &state, &[e]).await?;
+    assert!(metrics.num_scanned_files() == 1);
+
+    // Check pruning for null partitions. Since there are no record count statistics pruning cannot be done
```

Comment:
In Datafusion, null indicates that statistics for a particular column do not exist. Also null  counts are only used for the functions `is_null` and `is_not_null`. Further work is required to accommodate the edge case where all records for a columns are actually null. I imagine this can be accomplished by checking the null count statistic with a record count statistic.
I'd take a further look and raise an issue on their end.